src/index.test.ts

```diff
@@ -1483,15 +1555,16 @@ test('Should emit impression events on getVariant calls when impressionData is f
 });
 
 test('Should publish ready only when the first fetch was successful', async () => {
+    expect.assertions(3);
```

Comment:
its a bit confusing to see updates to existing tests that seems unrelated to the PR. Can you please elaborate on why you needed to modify this scenario?

Reply:
(the change looks reasonable tbh, but still would love to see your comments on them)

Reply:
In an early version of our implementation, we needed to update this test and actually we realized that the test was not working as it should so we fixed it, but we can remove that if you think that should not be part of this pull request.

Reply:
thanks for a clarification. I would prefer to have it as a separate PR if you do not mind. We can keep it here as well